src/libstore/build/local-derivation-goal.cc

```diff
@@ -2613,7 +2613,8 @@ DrvOutputs LocalDerivationGoal::registerOutputs()
             signRealisation(thisRealisation);
             worker.store.registerDrvOutput(thisRealisation);
         }
-        builtOutputs.emplace(thisRealisation.id, thisRealisation);
+        if (wantedOutputs.empty() || wantedOutputs.count(outputName))
```

Comment:
We have a function for this.
```suggestion
        if (wantOutput(outputName, wantedOutputs))
```